examples/plot_tuh_age_regression.py

```diff
@@ -0,0 +1,152 @@
+"""
+Age regression on TUH Abnormal EEG Dataset
+==========================================
+"""
+
+# Authors: Lukas Gemein <l.gemein@gmail.com>
+#
+# License: BSD-3
+import numpy as np
+import torch
+from skorch.callbacks import LRScheduler
+from skorch.helper import predefined_split
+
+from braindecode import EEGRegressor
+from braindecode.datautil import create_fixed_length_windows
+from braindecode.datasets import TUHAbnormal
+from braindecode.losses import CroppedLoss
+from braindecode.models import Deep4Net
+from braindecode.models import ShallowFBCSPNet
+from braindecode.models.util import to_dense_prediction_model, get_output_shape
+from braindecode.util import set_random_seeds
+from braindecode.datautil.transforms import transform_concat_ds
+
+model_name = "shallow"  # 'shallow' or 'deep'
+n_epochs = 5
+seed = 20200220
+
+input_time_length = 6000
+batch_size = 64
+cuda = torch.cuda.is_available()
+device = 'cuda' if cuda else 'cpu'
+if cuda:
+    torch.backends.cudnn.benchmark = True
+
+n_chans = 21
+n_classes = 1
+
+set_random_seeds(seed=seed, cuda=cuda)
+
+# initialize a model, transform to dense and move to gpu
+if model_name == "shallow":
+    model = ShallowFBCSPNet(
+        in_chans=n_chans,
+        n_classes=n_classes,
+        input_time_length=input_time_length,
+        n_filters_time=40,
+        n_filters_spat=40,
+        final_conv_length=35,
+    )
+    optimizer_lr = 0.000625
+    optimizer_weight_decay = 0
+elif model_name == "deep":
+    model = Deep4Net(
+        in_chans=n_chans,
+        n_classes=n_classes,
+        input_time_length=input_time_length,
+        n_filters_time=25,
+        n_filters_spat=25,
+        stride_before_pool=True,
+        n_filters_2=int(n_chans * 2),
+        n_filters_3=int(n_chans * (2 ** 2.0)),
+        n_filters_4=int(n_chans * (2 ** 3.0)),
+        final_conv_length=1,
+    )
+    optimizer_lr = 0.01
+    optimizer_weight_decay = 0.0005
+else:
+    raise ValueError(f'{model_name} unknown')
+
+new_model = torch.nn.Sequential()
+for name, module_ in model.named_children():
+    if "softmax" in name:
+        continue
+    new_model.add_module(name, module_)
+model = new_model
+
+if cuda:
+    model.cuda()
+
+to_dense_prediction_model(model)
+n_preds_per_input = get_output_shape(model, n_chans, input_time_length)[2]
+
+dataset = TUHAbnormal(
+    path="/data/schirrmr/gemeinl/tuh-abnormal-eeg/raw/v2.0.0/edf/train/",
```

Comment:
I see that this path does not exist in the CI, so that's why your PR is failing, any idea how to fix this?

Reply:
Well, a solution would be to not use the TUH dataset. For the purpose of this example I could fake data using <code>braindecode.util.create_mne_dummy_raw</code>.

Reply:
Don't we want to have nice plots and make an example of real data usage? If we use dummy input it is no longer a real example. I don't know what is the best solution, but possible from my perspective:
1. We use another smaller dataset on which we can perform regression (TUH dataset is too big to use it on CI? Am I right?)
2. We use dummy input and treat this as a not real-life example (other examples uses only small datasets, so they are not real-life too?).
3. We add functionality to TUHDataset to use only a subset of data for TUH dataset, so it works on CI.

Reply:
Yes you are right. It would then not be a real example anymore, just an example that shows how to generally do it.
1. I guess the main problem is, that although the TUH dataset is publicly available, it requires for registration. I think, we cannot reupload files without asking for permission. Maybe moabb could ask for integration in their data fetching.
2. Yes, that would be easiest for now.
3. We have this subset reading functionality already (see <code>subject_ids</code> https://github.com/braindecode/braindecode/blob/master/braindecode/datasets/datasets.py#L125-L126).

Reply:
yeah to me dummy for now seems best. I think making a real example out of it is sth that can be done at any point independently of other work